SEAPODYM_Data_Test.py

```diff
@@ -0,0 +1,94 @@
+from parcels import Particle, JITParticle, AdvectionRK4
+from parcels.grid import Grid
+from argparse import ArgumentParser
+from py import path
+import numpy as np
+
+
+def updateUserVars(particle, grid, time, dt):
+    for var in particle.user_vars.keys():
+        setattr(particle, var, getattr(grid, var)[time, particle.lon, particle.lat])
+
+
+if __name__ == "__main__":
+    p = ArgumentParser(description="""
+        Example of particle advection around an idealised peninsula""")
+    p.add_argument('mode', choices=('scipy', 'jit'), nargs='?', default='scipy',
+                   help='Execution mode for performing RK4 computation')
+    p.add_argument('-p', '--particles', type=int, default=10,
+                   help='Number of particles to advect')
+    p.add_argument('--profiling', action='store_true', default=False,
+                   help='Print profiling information after run')
+    p.add_argument('-l', '--location', default=None,
+                   help='Location folder path of input NetCDF files')
+    p.add_argument('-f', '--files', default=['SEAPODYM_PHYS.nc', 'SEAPODYM_PHYS.nc', 'SEAPODYM_HABITAT_depth5.nc', 'SEAPODYM_Density7.nc'],
+                   help='List of NetCDF files to load')
+    p.add_argument('-v', '--variables', default=['U', 'V', 'habitat', 'BIO'],
+                   help='List of field variables to extract, using PARCELS naming convention')
+    p.add_argument('-n', '--netcdf_vars', default=['u', 'v', 'habitat', 'skipjack_cohort_20021015_density'],
+                   help='List of field variable names, as given in the NetCDF file. Order must match --variables args')
+    p.add_argument('-d', '--dimensions', default=['lat', 'lon', 'time'],
+                   help='List of PARCELS convention named dimensions across which field variables occur')
+    p.add_argument('-m', '--map_dimensions', default=['lat', 'lon', 'time'],
+                   help='List of dimensions across which field variables occur, as given in the NetCDF files, to map to the --dimensions args')
+    p.add_argument('-o', '--output_file', default='SEAPODYM_Particles',
+                   help='Name of output NetCDF file for particle data')
+    args = p.parse_args()
```

Comment:
This setup assumes a whole lot of default files that are not (and should not be) part of the main repo. If you are free to upload down-sampled versions of the these, I think we should move the entire example to `parcels-examples`.

As far as I can see it also doesn't add any automated tests to our regressions test suite, and the newly added features should easily be testable with one of the existing tests. This demonstrates that we need to start caring about what is a test and should be in the main repo and what goes into the examples sub-repo. I'll create an issue to that effect.